vision/pipeline.py

```diff
@@ -212,47 +224,47 @@ def run(self, prev_state):
                                             :,
                                         ]
                                     )  # roll of module
-                                except:
-                                    flags.get_module_roll = False
-
-                                if flags.get_module_roll:
                                     # construct boundingbox for the module
                                     box = BoundingBox(bounds, ObjectType.MODULE)
                                     box.module_depth = depth  # float
                                     box.orientation = orientation + (
                                         roll,
                                     )  # x, y, z tilt
-
                                     bboxes.append(box)
+                                except:
+                                    flags.get_module_roll = False
 
         else:
             pass  # raise AttributeError(f"Unrecognized state: {state}")
 
-        self.vision_flags.append(flags)
+        time = datetime.datetime.now()
 
+        # You need to index vision_flags to see output, "self.vision_flags.get()[1]"
+        self.vision_flags.put(
+            (time, flags), self.PUT_TIMEOUT
+        )
+        await asyncio.sleep(0.01)
         ##
         self.vision_communication.put(
-            (datetime.datetime.now(), bboxes), self.PUT_TIMEOUT
+            (time, bboxes), self.PUT_TIMEOUT
         )
-
+        await asyncio.sleep(0.01)
         # uncomment to visualize blobs
         # from vision.common.blob_plotter import plot_blobs
         # plot_blobs(self.obstacle_finder.keypoints, color_image)
 
         return state
 
 
-def init_vision(vision_comm, flight_comm, video, runtime=100):
+async def init_vision(vision_comm, flight_comm, camera, state, runtime=100):
     """
     Alex, call this function - not run.
     """
-    pipeline = Pipeline(vision_comm, flight_comm, video)
-
-    state = "start"
-
-    for _ in range(runtime):
-        state = pipeline.run(state)
+    pipeline = Pipeline(vision_comm, flight_comm, camera)
 
+    warnings.filterwarnings("ignore")
```

Comment:
Oh, sorry, by init I mean the filter warnings should go in the `pipeline`'s member function `__init__()`